syntax = "proto3";

message InputEvent {
    int32 type = 1;     // 事件类型
    int32 code = 2;     // 事件码
    int32 value = 3;    // 事件值
}

message Start {};
message Stop {};

message InputGrabberParent {
    oneof payload {
        Start start = 2;
        Stop stop = 3;
    }
}

message InputGrabberChild {
    oneof payload {
        uint32 deviceType = 1;
        InputEvent inputEvent = 2;
    }
}

message InputEmittorParent {
    oneof payload {
        InputEvent inputEvent = 1;
    }
}
